[MM-66591] Channel Summarization - header icon pluggable + citation support

nickmisasi opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
This PR adds support for 2 things:

1. A new Channel Header Icon pluggable spot, next to the files button in the Channel Header. This houses the Agents Channel Actions button.
2. Support for inline-entity-link type in markdown processing, which parses site-url based links containing a query parameter `?view=citation` and renders them as a citation link, with a popover, and redux-powered click action (to prevent thrashing of react router)


#### Ticket Link
<!--
If applicable, please include both or either of the following links:

Fixes https://github.com/mattermost/mattermost/issues/XXX
Jira https://mattermost.atlassian.net/browse/MM-XXX
-->

#### Screenshots
<img width="553" height="71" alt="image" src="https://github.com/user-attachments/assets/1d9792a2-09b7-4539-8a15-4124cae69cdd" />
<img width="462" height="125" alt="image" src="https://github.com/user-attachments/assets/b4164dde-207f-45b7-98ec-2f8fc66cf654" />

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->
```release-note
None
```

hmhealey commented on webapp/channels/src/utils/message_html_to_component.tsx:
```diff
@@ -108,6 +112,76 @@ export default function messageHtmlToComponent(html: string, options: Options =
+            const url = node.attribs.href;
+            return url.includes('view=citation');
+        },
+        processNode: (node: any, _children: any, index?: number) => {
```
I don't think it's a good idea to split the logic of identifying and parsing the link between this file and `InlineEntityLink`.  `messageHtmlToComponent` is already quite large, even though most of the other cases in it don't include any logic for handling how the element, so I think it would be better for this to live in `InlineEntityLink`

hmhealey commented on webapp/channels/src/utils/message_html_to_component.tsx:
```diff
@@ -108,6 +112,76 @@ export default function messageHtmlToComponent(html: string, options: Options =
+            // Matches:
+            // POST: .../pl/<id>
+            // CHANNEL: .../<team>/channels/<channel>
+            // TEAM: .../<team>
```
Is there any chance that this will also detect something that looks like a team name but isn't? Will we run into any issues if something is incorrectly identified as a team name by this?

We really should've made the URL something like `SERVER/teams/TEAM/channels/CHANNEL` to avoid that sort of ambiguity

hmhealey commented on webapp/channels/src/components/post_view/post_list/post_list.tsx:
```diff
@@ -207,6 +207,11 @@ export default class PostList extends React.PureComponent<Props, State> {
             this.postsOnLoad(this.props.channelId);
         }
+
+        if (this.props.focusedPostId !== prevProps.focusedPostId) {
```
Could you combine this block and the above one? As this is currently written, it calls `postsOnLoad` twice when switching to a permalink in another channel

hmhealey commented on webapp/channels/src/components/post_view/post_list_virtualized/post_list_virtualized.tsx:
```diff
@@ -233,6 +233,20 @@ export default class PostList extends React.PureComponent<Props, State> {
             this.scrollStopAction = new DelayedAction(this.handleScrollStop);
         }
 
+        if (this.props.focusedPostId) {
```
What's this logic for? I know the permalink scrolling isn't great, but I'm wary of adding complexity to it without understanding how it interacts with the existing complexity

nickmisasi replied:
IIRC this was a similar bug to https://github.com/mattermost/mattermost/pull/34687#discussion_r2688733913 wrt jumping around a channel from a single summary. added a comment

hmhealey commented on webapp/channels/src/components/inline_entity_link/index.tsx:
```diff
@@ -0,0 +1,114 @@
+        case Constants.InlineEntityTypes.CHANNEL: {
+            // If we have a value (ID), use it directly
+            if (value) {
+                store.dispatch(loadIfNecessaryAndSwitchToChannelById(value));
```
I don't think this case is possible. `messageHtmlToComponent` never populates `value` for channels

hmhealey commented on webapp/channels/src/components/inline_entity_link/index.tsx:
```diff
@@ -0,0 +1,114 @@
+        }
+        case Constants.InlineEntityTypes.TEAM: {
+            if (value) {
+                const team = getTeam(state, value) || getTeamByName(state, value);
```
`messageHtmlToComponent` does populate `value` for team links, but that passes the team name as `value` instead of the team's ID.

Could you actually change the `value` prop to `postId` and only use it for the post link case? Calling it `value` is sure to confuse people working on this code in the future as it already seems to be doing

hmhealey commented on webapp/channels/src/components/inline_entity_link/index.tsx:
```diff
@@ -0,0 +1,114 @@
+                        store.dispatch(selectTeam(teamByName.id));
+                        store.dispatch(switchTeam(`/${teamByName.name}`));
+                    }
+                }
```
Could you clean up this logic a bit? `switchTeam` already dispatches `selectTeam` when passed the Team as a parameter, and we've already gotten the team by name above

hmhealey commented on webapp/channels/src/components/inline_entity_link/index.tsx:
```diff
@@ -0,0 +1,114 @@
+    };
+
+    const typeLabel = type.toLowerCase();
+    const tooltipText = `Go to ${typeLabel}`;
```
This needs to be translated, and we should have a translation string for each type of link since the translation may vary based on what's being linked to

nickmisasi commented:
@hmhealey I think I've addressed all the feedback, let me know if I missed anything. 

I added a few unit tests to test out the inline citations specifically. We have E2E tests on the Agents side of things: https://github.com/mattermost/mattermost-plugin-agents/pull/460
